package/MDAnalysis/analysis/encore/utils.py

```diff
@@ -0,0 +1,390 @@
+
+    `metadata` : dict
+            Metadata for the matrix (date of creation, name of author ...)
+    """
+
+    def __init__(self, size, metadata=None, loadfile=None):
+        """Class constructor.
+
+        Attributes
+        ----------
+
+            `size` : int or multiprocessing.SyncrhonizeArray
+        Size of the matrix (number of rows or columns). If an array is
+        provided instead, the size of the triangular matrix will be
+        calculated and the array copied as the matrix elements. Otherwise,
+        the matrix is just initialized to zero.
+
+            `metadata` : dict or None
+        Metadata dictionary. Used to generate the metadata attribute.
+
+            `loadfile` : str or None
+        Load the matrix from this file. All the attributes and data will
+        be determined by the matrix file itself (i.e. metadata will be
+        ignored); size has to be provided though.
+        """
+        self.metadata = metadata
+        self.size = size
+        if loadfile:
+            self.loadz(loadfile)
+            return
+        if type(size) == int:
+            self.size = size
+            self._elements = zeros((size + 1) * size / 2, dtype=float64)
+            return
+        if type(size) == SynchronizedArray:
+            self._elements = array(size.get_obj(), dtype=float64)
+            self.size = int((sqrt(1 + 8 * len(size)) - 1) / 2)
+            return
+        else:
+            raise TypeError
+
+    def __getitem__(self, args):
+        x, y = args
+        if x < y:
+            x, y = y, x
+        return self._elements[x * (x + 1) / 2 + y]
+
+    def __setitem__(self, args, val):
+        x, y = args
+        if x < y:
+            x, y = y, x
+        self._elements[x * (x + 1) / 2 + y] = val
+
+    def savez(self, fname):
+        """Save matrix in the npz compressed numpy format. Save metadata and
+        data as well.
+
+        Parameters
+        ----------
+
+        `fname` : str
+            Name of the file to be saved.
+        """
+        savez(fname, elements=self._elements, metadata=self.metadata)
+
+    def loadz(self, fname):
+        """Load matrix from the npz compressed numpy format.
+
+        Parameters
+        ----------
+
+        `fname` : str
+            Name of the file to be loaded.
+        """
+        loaded = load(fname)
+
+        if loaded['metadata'].shape != ():
+            if loaded['metadata']['number of frames'] != self.size:
+                raise TypeError
+            self.metadata = loaded['metadata']
+        else:
+            if self.size*(self.size-1)/2+self.size != len(loaded['elements']):
+                raise TypeError
+        self._elements = loaded['elements']
+
+    def change_sign(self):
+        """
+        Change sign of each element of the matrix
+        """
+        for k, v in enumerate(self._elements):
+            self._elements[k] = -v
+
+
+class ParallelCalculation:
```

Comment:
Eventually we'd like to have a unified approach to handling parallel analysis (PR #618 is going in this direction and there's also some discussion in the proposed MDAnalysis.analysis intefrace #719. But for right now, keep your implementation and we'll clean up later. You might even want to chime in in the discussion and say what you need.